app/src/main/java/com/github/se/eventradar/model/repository/FirebaseEventRepository.kt

```diff
@@ -0,0 +1,132 @@
+package com.github.se.eventradar.model.repository
+
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.getEventCategory
+import com.github.se.eventradar.model.event.getEventTicket
+import com.github.se.eventradar.model.getLocation
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import kotlinx.coroutines.tasks.await
+
+class FirebaseEventRepository : EventRepository {
```

Comment:
imp: I think we should add @Inject constructor here so that Hilt knows how to provide Instances of FirebaseEventRepository (same with the FIrebaseUserRepository, MockEventRepository and MockUserRepository)

Reply:
> nit: maybe we could rename this to FIrebaseEventRepositoryImpl too for redability? (Apologies for the many nits on naming)

In my opinion, adding Impl doesn't make it any more readable. Adding a descriptive word or two at the beginning (like `Firebase` or `Mock`) goes a much farther way in helping the developer understand the purpose of the class.